app/keepers/keepers.go

```diff
@@ -307,6 +310,14 @@ func (appKeepers *AppKeepers) InitNormalKeepers(
 	)
 	appKeepers.TxFeesKeeper = &txFeesKeeper
 
+	tokenFactoryKeeper := tokenfactorykeeper.NewKeeper(
+		appCodec,
+		appKeepers.keys[tokenfactorytypes.StoreKey],
+		appKeepers.AccountKeeper,
+		appKeepers.BankKeeper,
```

Comment:
This needs a `WithMintCoinsRestriction`

Reply:
See: https://github.com/osmosis-labs/cosmos-sdk/blob/osmosis-main/x/bank/keeper/keeper.go#L147-L163 for reference

Reply:
Added